src/scene/base.rs

```diff
@@ -727,113 +752,184 @@ impl Base {
+                if let Some(sender) = self.script_message_sender.as_ref() {
+                    Log::verify(sender.send(NodeScriptMessage::DestroyScript {
+                        script,
+                        handle: self.self_handle,
+                    }));
+                } else {
+                    Log::warn(format!(
+                        "There is a script instance on a node {}, but no message sender. \
                     The script won't be correctly destroyed!",
-                    self.name(),
-                ))
+                        self.name(),
+                    ));
+                }
+            }
+        }
+    }
+
+    /// Removes all assigned scripts from scene node
+    fn remove_all_scripts(&mut self) {
+        for script in self.scripts.iter_mut() {
+            // Send script to the graph to destroy script instances correctly.
+            if let Some(script) = script.take() {
+                if let Some(sender) = self.script_message_sender.as_ref() {
+                    Log::verify(sender.send(NodeScriptMessage::DestroyScript {
+                        script,
+                        handle: self.self_handle,
+                    }));
+                } else {
+                    Log::warn(format!(
+                        "There is a script instance on a node {}, but no message sender. \
+                            The script won't be correctly destroyed!",
+                        self.name.as_str()
+                    ));
+                }
             }
         }
     }
 
-    /// Sets new script for the scene node.
+    /// Sets a new script for the scene node by index
     #[inline]
-    pub fn set_script(&mut self, script: Option<Script>) {
-        self.remove_script();
-        self.script = script;
+    pub fn set_script(&mut self, index: usize, script: Option<Script>) {
+        self.remove_script(index);
+        if index < self.scripts.len() {
+            self.scripts[index] = ScriptWrapper(script);
+        }
         if let Some(sender) = self.script_message_sender.as_ref() {
-            if self.script.is_some() {
+            if self.scripts[index].is_some() {
                 Log::verify(sender.send(NodeScriptMessage::InitializeScript {
                     handle: self.self_handle,
                 }));
             }
         }
     }
 
-    fn set_script_internal(&mut self, script: Option<Script>) -> Option<Script> {
-        std::mem::replace(&mut self.script, script)
+    /// Adds a new script to the scene node.
+    #[inline]
+    pub fn add_script(&mut self, script: Script) {
+        self.scripts.push(ScriptWrapper(Some(script)));
+        if let Some(sender) = self.script_message_sender.as_ref() {
+            Log::verify(sender.send(NodeScriptMessage::InitializeScript {
+                handle: self.self_handle,
+            }));
+        }
+    }
+
+    fn set_scripts_internal(&mut self, (index, script): (usize, Option<Script>)) -> Option<Script> {
```

Comment:
This method is used in reflection-based access to `scripts` field and its signature must be `fn set_scripts_internal(&mut self, scripts: Vec<ScriptWrapper>) -> Vec<ScriptWrapper>`